### PR TITLE
Obscuroscan#2323

### PR DESCRIPTION
**Pull request by Jennievon**
### Why this change is needed

Please provide a description and a link to the underlying ticket

https://github.com/obscuronet/obscuro-internal/issues/2323

### What changes were made as part of this PR

Please provide a high-level list of the changes made

- Revamp of Obscuroscan

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: tools/obscuroscan_v2/frontend/src/components/helper/ShortenedHash.vue
Comment (otherview) [inline tools/obscuroscan_v2/frontend/src/components/helper/ShortenedHash.vue:32]:
```suggestion
      const start = this.hash.substring(0, 12);
      return `${start}...`;
```

File: tools/obscuroscan_v2/frontend/src/components/helper/Timestamp.vue
Comment (otherview) [inline tools/obscuroscan_v2/frontend/src/components/helper/Timestamp.vue:4]: Why `unixTimestampSeconds * 1000` ?

File: tools/obscuroscan_v2/frontend/src/main.js
Comment (otherview) [inline tools/obscuroscan_v2/frontend/src/main.js:23]: Sweet ! Nice one on the comment 👏

File: tools/obscuroscan_v2/frontend/src/views/NavbarView.vue
Comment (otherview) [inline tools/obscuroscan_v2/frontend/src/views/NavbarView.vue:52]:
```suggestion
  background: white;
  padding-left: 12px;
  padding-right: 8px;
```

File: tools/obscuroscan_v3/frontend/README.md
Comment (otherview) [inline tools/obscuroscan_v3/frontend/README.md:1]: Should this file be updated ? Would it make sense to clean up the file and remove stuff like the deploy instructions ?

**Review by otherview — COMMENTED** [review]
Looks good ! Added a few comments on the v2. 
Can this broken down into 2 PRs ? Updates to the v2 and adding the v3 ?

**Comment by BedrockSquirrel** [comment]
This feels like a good opportunity to get rid of the v2/v3 thing - wdyt @otherview ?

For example, in this PR we could:
1. Delete the contents of `/tools/obscuroscan` (think they can safely be deleted as unused)
2. Move the contents of this PR to `/tools/obscuroscan/frontend`

And in a follow-up PR:
1. Move the contents of `/tools/obscuroscan_v2/backend` to `tools/obscuroscan/backend`
2. Update deployment scripts to deploy the f/e and backend from new locations
3. Delete `/tools/obscuroscan_v2`

File: tools/obscuroscan_v3/frontend/public/next.svg
Comment (BedrockSquirrel) [inline tools/obscuroscan_v3/frontend/public/next.svg:1]: Can delete this and the vercel logos I guess.

File: tools/obscuroscan_v3/frontend/src/hooks/useBatches.ts
Comment (BedrockSquirrel) [inline tools/obscuroscan_v3/frontend/src/hooks/useBatches.ts:9]: Not sure we're going to want this to always be polling btw. I changed it on main to only poll for the front page because we don't want the table page batches to constantly scroll off screen (remember they can come as fast as 1 per sec).

**Review by BedrockSquirrel — COMMENTED** [review]
This is awesome work, and an absolute sh**ton of it!

I'm not familiar with typescript, nextjs or axios so apologies in advance for my ignorant questions. I was able to run it up locally and it's working great from a local env point of view. 

Also please feel free to just ignore any comments/nitpicks that you disagree with or are not high priority for now, I just wanted to note down things as I saw them.


I had a couple of high-level questions:

- do you see most of the `/components` stuff being extracted to be re-used for all your obscuro frontend stuff? Seems like only the `/modules` are obscuroscan specific.

- did you make any 'controversial' decisions in the frameworks/organisation/setup of this project? Like stuff you would feel the need to justify to an experienced front-end developer? Or do you feel that most of the stuff here is standard/best-practices? Not looking to second-guess those decisions, but just interested for my own learning (and useful to know in case we're ever googling issues).



And here's the nitpicks (feel free to ignore, I'm sure they're all on your list):
- we need to think about what headline dashboard data we care about, ideally want 4 rather than 5 to make that layout look its best I guess?
- the spacing on the navbar links is inconsistent for the ones with submenus I think, looks bit weird (and be cool to have hover for those submenus)
- it's annoying how the 'less than one minute ago' time description buggers the table layout, maybe it could use seconds or be shortened or something
- feel like the header/footer shouldn't be changing based on the width of the page content (like when you switch tabs it jumps around)
- I guess the button next to the dark theme toggle is a connect wallet button? Didn't feel super obvious to me so maybe worth a text label too

Anyway, I'm going to stop wasting your time by bike-shedding a work-in-progress, just needed to scratch that itch haha.

Love the copy buttons and the Finality icons and stuff. Tables look neat. Also the footer and the social icons are super nice.

Definitely think we should get this in asap so you can start iterating and we can get it deployed for people to play with.

**Comment by BedrockSquirrel** [comment]
Meant to say as well, the code organisation all seemed very logical to my untrained eye.

I don't fully grok the `/pages` stuff, I think you might need to do a technical show and tell at some point to give us a high-level walkthrough of this stuff.

But I loved the way the api code was kept separate and the component lib taking shape, and the tailwind stuff is so much nicer than giant filthy css files and conflicting class names.

File: tools/obscuroscan_v3/frontend/components/modules/batches/batch-details.tsx
Comment (Jennievon) [reply tools/obscuroscan_v3/frontend/components/modules/batches/batch-details.tsx:1]: Still trying to structure it. I'll push an update for it.

File: tools/obscuroscan_v3/frontend/public/next.svg
Comment (Jennievon) [reply tools/obscuroscan_v3/frontend/public/next.svg:1]: Sure!

File: tools/obscuroscan_v3/frontend/src/hooks/useBatches.ts
Comment (Jennievon) [reply tools/obscuroscan_v3/frontend/src/hooks/useBatches.ts:9]: Yeah, I used 5 seconds. However, we can take it out and have a refresh button on all the tables.

**Comment by Jennievon** [comment]
> Meant to say as well, the code organisation all seemed very logical to my untrained eye.
> 
> I don't fully grok the `/pages` stuff, I think you might need to do a technical show and tell at some point to give us a high-level walkthrough of this stuff.
> 
> But I loved the way the api code was kept separate and the component lib taking shape, and the tailwind stuff is so much nicer than giant filthy css files and conflicting class names.

Thank you. I appreciate the comment! ✨

**Comment by Jennievon** [comment]
> This is awesome work, and an absolute sh**ton of it!
> 
> I'm not familiar with typescript, nextjs or axios so apologies in advance for my ignorant questions. I was able to run it up locally and it's working great from a local env point of view.
> 
> Also please feel free to just ignore any comments/nitpicks that you disagree with or are not high priority for now, I just wanted to note down things as I saw them.
> 
> I had a couple of high-level questions:
> 
> * do you see most of the `/components` stuff being extracted to be re-used for all your obscuro frontend stuff? Seems like only the `/modules` are obscuroscan specific.
> * did you make any 'controversial' decisions in the frameworks/organisation/setup of this project? Like stuff you would feel the need to justify to an experienced front-end developer? Or do you feel that most of the stuff here is standard/best-practices? Not looking to second-guess those decisions, but just interested for my own learning (and useful to know in case we're ever googling issues).
> 
> And here's the nitpicks (feel free to ignore, I'm sure they're all on your list):
> 
> * we need to think about what headline dashboard data we care about, ideally want 4 rather than 5 to make that layout look its best I guess?
> * the spacing on the navbar links is inconsistent for the ones with submenus I think, looks bit weird (and be cool to have hover for those submenus)
> * it's annoying how the 'less than one minute ago' time description buggers the table layout, maybe it could use seconds or be shortened or something
> * feel like the header/footer shouldn't be changing based on the width of the page content (like when you switch tabs it jumps around)
> * I guess the button next to the dark theme toggle is a connect wallet button? Didn't feel super obvious to me so maybe worth a text label too
> 
> Anyway, I'm going to stop wasting your time by bike-shedding a work-in-progress, just needed to scratch that itch haha.
> 
> Love the copy buttons and the Finality icons and stuff. Tables look neat. Also the footer and the social icons are super nice.
> 
> Definitely think we should get this in asap so you can start iterating and we can get it deployed for people to play with.

Thank you for the thorough review and your questions. I appreciate your attention to detail, and I'll address each point:
1. The structure of the `/components` directory aims for a modular and reusable approach. Common components that can be reused across different parts of the application are placed there. The `/modules` directory, on the other hand, contains components more specific to the Obscuroscan application.
2. The decisions made in terms of frameworks, organisation & setup align with standard practices for React/Next.js projects. There might be variations in personal preferences, but overall, the structure is in line with best practices. If there are specific concerns or areas you'd like more clarification on, please let me know.

As for your "nitpicks":

1. It might be worth discussing and prioritising which metrics are most valuable for users.
2. I adjusted the spacing for links with submenus to ensure a more uniform appearance in a subsequent commit I made.
4. The issue with the "less than one minute ago" time description affecting table layout is noted. I'll explore options like using seconds or shortening the description for a better fit.
5. I'll work on refining the layout to prevent jumps during tab switches.
6. The text would typically be hidden and only the icon left for smaller screens, however, I can explore other options for this.

I appreciate your feedback. I'll address the mentioned points and look forward to your continued feedback as we aim to get this deployed for user testing soon.

To note: We're using a component library to ease reusability and consistency. They can be modified as needed too. I'll update the README file once I get the remaining features. Also, would we have API endpoints for single transactions and blocks, as we do for batches?

File: tools/obscuroscan_v3/frontend/src/hooks/useBatches.ts
Comment (BedrockSquirrel) [reply tools/obscuroscan_v3/frontend/src/hooks/useBatches.ts:9]: Ok great, yeah we can figure that out when we sort the pagination and stuff.

**Comment by BedrockSquirrel** [comment]
Thanks, that all sounds good to me. 

> Also, would we have API endpoints for single transactions and blocks, as we do for batches?
I'll create a ticket for this, should be able to get it for you this sprint but don't let it block this PR.

File: tools/obscuroscan_v3/frontend/README.md
Comment (otherview) [inline tools/obscuroscan_v3/frontend/README.md:40]: We wouldn't be deploying on vercel. While it's possible to become open sourced in the future, let's keep the docs tight. 
```suggestion

```

Comment (otherview) [reply tools/obscuroscan_v3/frontend/README.md:40]: Can a section be added with the folder structure and what is expected in each folder ? I feel like it can be annoying but will certainly help review and set up our development rails

File: tools/obscuroscan_v3/frontend/api/batches.ts
Comment (otherview) [inline tools/obscuroscan_v3/frontend/api/batches.ts:1]: Should the `api` be inside the `src` folder ?

File: tools/obscuroscan_v3/frontend/api/batches.ts
Comment (otherview) [inline tools/obscuroscan_v3/frontend/api/batches.ts:20]: These seem to be methods created to be imported into other parts of the code.
It might be confusing as `getBatches` is a pretty common name. What if it was something like `getBatchesAPI`.

File: tools/obscuroscan_v3/frontend/api/batches.ts
Comment (otherview) [inline tools/obscuroscan_v3/frontend/api/batches.ts:22]: Would apply the same explicit naming logic across the folder.

File: tools/obscuroscan_v3/frontend/api/batches.ts
Comment (otherview) [inline tools/obscuroscan_v3/frontend/api/batches.ts:19]: Nit: no need for the data var - Can probably just return the awaited promise instead ?

File: tools/obscuroscan_v3/frontend/components/date-range-picker.tsx
Comment (otherview) [inline tools/obscuroscan_v3/frontend/components/date-range-picker.tsx:1]: ooc, is there a reason for components to be outside of `src` ?

File: tools/obscuroscan_v3/frontend/components/modules/batches/data.tsx
Comment (otherview) [inline tools/obscuroscan_v3/frontend/components/modules/batches/data.tsx:1]: These seem to be constats yeah ? If so can the file to be renamed to something as such ? It's a small nit but data is super generic, would be great to have it locked down more imo.

File: tools/obscuroscan_v3/frontend/components/modules/batches/index.tsx
Comment (otherview) [inline tools/obscuroscan_v3/frontend/components/modules/batches/index.tsx:1]: Can be removed

File: tools/obscuroscan_v3/frontend/components/modules/blocks/columns.tsx
Comment (otherview) [inline tools/obscuroscan_v3/frontend/components/modules/blocks/columns.tsx:6]:
```suggestion
import { labels, statuses } from "./constants";
```
🚀

File: tools/obscuroscan_v3/frontend/components/modules/blocks/data.tsx
Comment (otherview) [inline tools/obscuroscan_v3/frontend/components/modules/blocks/data.tsx:1]: `constants.tsx` ?

File: tools/obscuroscan_v3/frontend/components/modules/common/connect-wallet.tsx
Comment (otherview) [inline tools/obscuroscan_v3/frontend/components/modules/common/connect-wallet.tsx:17]: Ignorance of mine, but whats the `<> ` ?

File: tools/obscuroscan_v3/frontend/components/modules/personal/data.tsx
Comment (otherview) [inline tools/obscuroscan_v3/frontend/components/modules/personal/data.tsx:1]: constants

File: tools/obscuroscan_v3/frontend/pages/batches/[batch].tsx
Comment (otherview) [inline tools/obscuroscan_v3/frontend/pages/batches/[batch].tsx:1]: Why the use of [ ] ?

File: tools/obscuroscan_v3/frontend/src/hooks/useBatches.ts
Comment (otherview) [inline tools/obscuroscan_v3/frontend/src/hooks/useBatches.ts:5]: While I understand `useBatches` naming can it be updated so it's clear what it does ?

File: tools/obscuroscan_v3/frontend/src/hooks/useBlocks.ts
Comment (otherview) [inline tools/obscuroscan_v3/frontend/src/hooks/useBlocks.ts:5]: Similar name prepending here

File: tools/obscuroscan_v3/frontend/src/routes/index.ts
Comment (otherview) [inline tools/obscuroscan_v3/frontend/src/routes/index.ts:1]: This seems to be a constants file and the router has a single function ? 
Do the consts get used without the `pathToUrl` func ?

File: tools/obscuroscan_v3/frontend/src/types/interfaces/index.ts
Comment (otherview) [inline tools/obscuroscan_v3/frontend/src/types/interfaces/index.ts:1]: ooc whats the use of interfaces ?

**Review by otherview — COMMENTED** [review]
Really good job ! A bit large of a PR which makes it harder to into the details.

File: tools/obscuroscan_v3/frontend/pages/batches/[batch].tsx
Comment (Jennievon) [reply tools/obscuroscan_v3/frontend/pages/batches/[batch].tsx:1]: It's a dynamic page. {URL}/batches/{hash}

**Review by otherview — APPROVED** [review]

**Comment by otherview** [comment]
Quick update - This was discussed offline and it will be added a new Dockerfile for the FE v3 and the new v3 workflow as the last items before merging to main.

File: .github/workflows/manual-deploy-obscuro-scan-3.yml
Comment (otherview) [inline .github/workflows/manual-deploy-obscuro-scan-3.yml:61]: Let's change this dns so it doesnt collide with the existing one
```suggestion
          dns-name-label: ${{ github.event.inputs.testnet_type }}-obscuro-scan-v3
```

File: .github/workflows/manual-deploy-obscuro-scan-3.yml
Comment (otherview) [inline .github/workflows/manual-deploy-obscuro-scan-3.yml:63]: Similar with the instance name
```suggestion
          name: ${{ github.event.inputs.testnet_type }}-fe-obscuro-scan-v3
```

File: tools/obscuroscan_v3/frontend/Dockerfile
Comment (otherview) [inline tools/obscuroscan_v3/frontend/Dockerfile:7]: Probably no need for the http server if next has a a prod server itself
```suggestion
RUN npm install
```

**Review by otherview — APPROVED** [review]

**Review by otherview — APPROVED** [review]

**Pull request merged**
